flink-streaming-java/src/test/java/org/apache/flink/streaming/timestamp/TimestampITCase.java

```diff
@@ -157,13 +157,6 @@ public void testWatermarkPropagation() throws Exception {
 					Assert.fail("Wrong watermark.");
 				}
 			}
-			if (!CustomOperator.finalWatermarks[i].get(CustomOperator.finalWatermarks[i].size() - 1).equals(new Watermark(Long.MAX_VALUE))) {
```

Comment:
Let's add a test here that validates that this `Long.MAX_VALUE` watermark is not present. Something like `assertEquals(NUM_WATERMARKS, CustomOperator.finalWatermarks[i].size())`